backends/ref/ceed-ref-operator.c

```diff
@@ -239,6 +222,179 @@ static int CeedOperatorApply_Ref(CeedOperator op, CeedVector invec,
       CeedChk(ierr);
     }
   }
+  return 0;
+}
+
+// Input basis action
+static inline int CeedOperatorInputBasis_Ref(CeedInt e, CeedInt Q,
```

Comment:
I did some substantial rearrangement, breaking the operator code into chunks for reuse.